src/main/java/edu/KaylaKornelis/advancedjava/Assignment4/BasicStockService.java

```diff
@@ -0,0 +1,101 @@
+package edu.KaylaKornelis.advancedjava.Assignment4;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import javax.validation.constraints.NotNull;
+import org.apache.http.annotation.Immutable;
+
+/**
+ *  This class implements the StockService interface for getting stock data.
+ */
+@Immutable
+public class BasicStockService implements StockService {
+  
+    private Date dateRecorded;
+    private BigDecimal stockPrice;
```

Comment:
if you initialized you these variables it would work otherwise you are creating a StockQuote with Null values which you don't want to do.

Reply:
initialized these variables